retry/retry.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package retry
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+const (
+	// defaultTimeout is a default timeout that is used
+	// when a timeout provided to the DialWithAttempts function is less or equal to zero.
+	defaultTimeout = time.Millisecond * 10
+
+	// defaultRetries is a default retries, that given to dial to vitess grpc server.
+	defaultRetries = 3
```

Comment:
Let's move these two values to the `config` package (and create a new config field `RetryTimeout time.Duration`).
And let's also rename them this way: `defaultRetries` -> `defaultMaxRetries` and `defaultTimeout` -> `defaultRetryTimeout`.

Reply:
And I think that the `defaultRetryTimeout` should be a bit longer for our case. Let's make it `time.Second`.